plugins/module_utils/foreman_helper.py

```diff
@@ -369,6 +374,13 @@ def ensure_entity(self, resource, desired_entity, current_entity, params=None, s
                 changed, updated_entity = self._delete_entity(resource, current_entity, params, synchronous)
         else:
             self.fail_json(msg='Not a valid state: {0}'.format(state))
+
+        if self._after == {}:
+            self._after = updated_entity
+        else:
+            # ensure_entity was called multiple times, we can't be sure which entity is the real one
```

Comment:
This can only happen if there are sub-entities, right? Do you think, we can add a `subentity_path`, where the diff can be placed? It would of course complicate the before/after tracking.

Reply:
params and stuff, and modules that support `affect_multiple`